File: src/sdk-client/Windows.Midi.Messages/Types/ChannelVoice/Base/Midi1ChannelVoiceMessage.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See LICENSE in the project root for license information.

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Microsoft.Windows.Midi.Messages.Types.ChannelVoice
{
    public abstract class Midi1ChannelVoiceMessage
    {
        public byte Group {  get; set; }
        public byte Status { get; set; }
        public byte Channel { get; set; }

    }
}

File: src/sdk-client/zzold/Microsoft.Windows.Midi.Native/Midi1ChannelVoiceMessage.cpp
// ------------------------------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See LICENSE in the GitHub project root for license information.
// ------------------------------------------------------------------------------------------------

#define WINDOWSMIDISERVICES_EXPORTS

#include "WindowsMidiServicesUtility.h"
#include "WindowsMidiServicesMessages.h"

namespace Microsoft::Windows::Midi::Messages
{

	const uint8_t Midi1ChannelVoiceMessage::getOpcode()
	{
		return Utility::MostSignificantNibble(Byte[1]);
	}

	const MidiChannel Midi1ChannelVoiceMessage::getChannel()
	{
		return (MidiChannel)Utility::LeastSignificantNibble(Byte[1]);
	}

	void Midi1ChannelVoiceMessage::setChannel(const MidiChannel value)
	{
		Utility::SetLeastSignificantNibble(Byte[1], value);
	}


}